lib/ex_unit/test/ex_unit/capture_io_test.exs

```diff
@@ -86,6 +86,102 @@ defmodule ExUnit.CaptureIOTest do
            end) == "a"
   end
 
+  test "async capture_io works with put chars to stderr" do
+    me = self()
+
+    Enum.each(1..6, fn num ->
+      spawn(fn ->
+        captured =
+          capture_io(:stderr, fn ->
+            :io.put_chars(:standard_error, "#{num}\n")
+            Process.sleep(10)
+            :io.put_chars(:standard_error, "#{num + 10}\n")
+          end)
+
+        send(me, captured)
+      end)
+
+      Process.sleep(1)
```

Comment:
Instead of sleeps, please use explicit messaging. Something like:

```elixir
pids =
  for num <- 1..6 do
    spawn(fn ->
      captured =
        capture_io(:stderr, fn ->
          :io.put_chars(:standard_error, "#{num}\n")
          send(me, :waiting)
          assert_receive :continue
          :io.put_chars(:standard_error, "#{num + 10}\n")
        end)

      send(me, {:captured, captured})
    end)

    assert_receive :waiting
  end

Enum.each(pid, &send(&1, :continue))
```

Similar in the other tests.